addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/handler/OnkyoHandler.java

```diff
@@ -338,7 +376,13 @@ public void statusUpdateReceived(String ip, EiscpMessage data) {
                  * ZONE 1
                  */
                 case POWER:
-                    updateState(CHANNEL_POWER, convertDeviceValueToOpenHabState(data.getValue(), OnOffType.class));
+                    State state = convertDeviceValueToOpenHabState(data.getValue(), OnOffType.class);
+                    updateState(CHANNEL_POWER, state);
```

Comment:
This is still somehow agly to me, even it’s work fine.

Refresh interval most probably is not relevant here, so could this be changed e.g. following

```
if (isPowerSwitchedOn(newPowerState)) {
    doPowerOnCheck();
}
```

Where method isPowerSwitchedOn do all lasPowerCheck staff and doPowerOnCheck method trigger all queries (currently only info query)

Reply:
@paulianttila I will update the Zone 1 Power code with what you’ve recommended above. Do you think the switched on check should be put in the Zone 2/3/4 power on section as well or is that excessive?

Reply:
@paulianttila can you comment here? I would like to get this PR merged before the bnd migration.

Reply:
Okay I've tidied that part of the code up. Power On checks affects all Zones for now.